app/src/test/groovy/pl/futurecollars/invoicing/db/memory/InMemoryDatabaseTest.groovy

```diff
@@ -0,0 +1,94 @@
+package pl.futurecollars.invoicing.db.memory
+
+import pl.futurecollars.invoicing.db.Database
+import pl.futurecollars.invoicing.model.Invoice
+import spock.lang.Specification
+
+import static pl.futurecollars.invoicing.helpers.TestHelpers.invoice
+
+class InMemoryDatabaseTest extends Specification {
```

Comment:
nie za bardzo rozumiem dlaczego tutaj zrobiłeś coś takiego, przecież twoja klasa InMemoryDatabaseIntegrationTest rozszerza AbstractDatabaseTest a w nim są w zasadzie wszystkie metody jakie tutaj zaimplementowałeś, wiec de facto zrobiłeś 2x takie same testy? dobrze mówie?

Reply:
Zresztą zobacz na pkt 11) z pracy domowej: Do not duplicate tests - if some tests can be common for InMemoryDatabase and FileBasedDatabase then you should extract it to the common class.